jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/DisplayMessageActivity.java

```diff
@@ -0,0 +1,29 @@
+package com.e_conomic.jonfirstapp;
+
+import android.content.Intent;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+public class DisplayMessageActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_display_message);
+
+        // Grab the intent that started this activity.
+        Intent intent = getIntent();
+        // Get the message to be shown.
+        String message = intent.getStringExtra(MainActivity.EXTRA_MESSAGE);
+        // Create the text view and set the message.
+        TextView textView = new TextView(this);
```

Comment:
Very cool that you did this in code, but for the purpose of this exercise we should probably stick to referencing a `TextView` that is already written in the xml.